src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -175,7 +175,12 @@ public void InstallWorkloads(IEnumerable<WorkloadId> workloadIds, bool skipManif
             {
                 // Update currently installed workloads
                 var installedWorkloads = _workloadInstaller.GetWorkloadInstallationRecordRepository().GetInstalledWorkloads(_sdkFeatureBand);
-                workloadIds = workloadIds.Concat(installedWorkloads).Distinct();
+                var previouslyInstalledWorkloads = installedWorkloads.Intersect(workloadIds);
+                if (previouslyInstalledWorkloads.Any())
+                {
+                    throw new GracefulException(string.Format(LocalizableStrings.WorkloadAlreadyInstalled, string.Join(" ", previouslyInstalledWorkloads)));
```

Comment:
should we join the string on ", " instead so it reads easier?

Reply:
I think it's standard to use spaces since commas don't always translate well when these strings are localized